support/cas-server-support-redis-ticket-registry/src/test/java/org/apereo/cas/ticket/registry/RedisServerTicketRegistryTests.java

```diff
@@ -288,6 +290,49 @@ void verifyDifferentLoginSamePrincipal() throws Throwable {
         }
     }
 
+    @Nested
+    @SpringBootTest(
+            classes = {
+                    CasRedisCoreAutoConfiguration.class,
+                    CasRedisTicketRegistryAutoConfiguration.class,
+                    BaseTicketRegistryTests.SharedTestConfiguration.class
+            }, properties = {
+            "cas.ticket.tgt.core.only-track-most-recent-session=false",
+            "cas.ticket.registry.redis.host=localhost",
+            "cas.ticket.registry.redis.port=6379",
+            "cas.ticket.registry.redis.crypto.enabled=false"
+    })
+    class TrackAllSessionsTests {
+        @Autowired
+        @Qualifier(TicketRegistry.BEAN_NAME)
+        private TicketRegistry ticketRegistry;
+
+        @Autowired
+        @Qualifier("ticketRedisTemplate")
+        private CasRedisTemplate<String, RedisTicketDocument> ticketRedisTemplate;
+
+        @Test
+        void verifyDifferentLoginSamePrincipal() throws Throwable {
+            val principalId = UUID.randomUUID().toString();
+            for (int i = 0; i < 5; i++) {
+                addTicketAndWait(principalId);
+            }
+            assertEquals(1, ticketRedisTemplate.boundZSetOps("CAS_PRINCIPAL:" + principalId).size());
```

Comment:
Please use `RedisCompositeKey`'s relevant methods for building keys.

Reply:
OK. I will change that.